### PR TITLE
feat: add BN254 final exponentiation check with output

### PR DESCRIPTION
**Pull request by ivokub**
# Description

In some cases we are interested to obtain the value if the final exponentiation resulted in 1 or not instead of asserting. Refactor and add such method.

With this, also renamed `FinalExpCheck` to `AssertFinalExpCheck` to better distinguish different methods.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How has this been tested?

- [x] TestIsMillerLoopAndFinalExpCircuitTestSolve
- [x] TestMillerLoopAndMulTestSolve

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I did not modify files generated from templates
- [x] `golangci-lint` does not output errors locally
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: std/algebra/emulated/sw_bn254/pairing.go
Comment (yelhousni) [inline std/algebra/emulated/sw_bn254/pairing.go:686]: `millerLoopAndFinalExpResult`

File: std/algebra/emulated/sw_bn254/pairing.go
Comment (yelhousni) [inline std/algebra/emulated/sw_bn254/pairing.go:816]: `IsMillerLoopAndFinalExpOne`

File: std/algebra/emulated/sw_bn254/pairing.go
Comment (yelhousni) [inline std/algebra/emulated/sw_bn254/pairing.go:832]: `AssertMillerLoopAndFinalExpIsOne`

**Review by yelhousni — APPROVED** [review]
LGTM! just made same renaming suggestions.

File: std/algebra/emulated/sw_bn254/pairing.go
Comment (ivokub) [reply std/algebra/emulated/sw_bn254/pairing.go]: Renamed

File: std/algebra/emulated/sw_bn254/pairing.go
Comment (ivokub) [reply std/algebra/emulated/sw_bn254/pairing.go]: Renamed

File: std/algebra/emulated/sw_bn254/pairing.go
Comment (ivokub) [reply std/algebra/emulated/sw_bn254/pairing.go]: Renamed

**Comment by ivokub** [comment]
Applied the review suggestions + added documentation for `IsMillerLoopAndFinalExpOne`. Will wait for CI before merge.

**Pull request merged**
